### PR TITLE
Add support for S3 inventory

### PR DESCRIPTION
**Pull request by KirillDavydov**

### CODE REVIEW

File: src/erlcloud_s3.erl [hunk 1]
```diff
@@ -1227,6 +1232,228 @@ set_bucket_attribute(BucketName, AttributeName, Value, Config)
...
+    make_list_inventory_result(
+        Truncated,
+        ContinuationToken,
+        NextContinuationToken,
+        UpdatedResponse
+    ).
+
+make_list_inventory_result(false, _, _, Result)->
+    {ok, Result};
+make_list_inventory_result(_, ContinuationToken, [], Result)->
```
Comment (motobob) [inline src/erlcloud_s3.erl]: nope. see http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketListInventoryConfigs.html
only two clauses are here

File: src/erlcloud_s3.erl [hunk 2]
```diff
@@ -1227,6 +1232,228 @@ set_bucket_attribute(BucketName, AttributeName, Value, Config)
...
+
+    InventoryId = proplists:get_value(id, Inventory),
+    XmlInventory = encode_inventory(Inventory),
+    put_bucket_inventory(BucketName, InventoryId, list_to_binary(XmlInventory), Config).
+
+-spec put_bucket_inventory(string(), string(), binary(), aws_config())
+        -> ok | {error, Reason::term()}.
+put_bucket_inventory(BucketName, InventoryId, XmlInventory, #aws_config{} = Config)
+    when is_list(BucketName), is_list(InventoryId), is_binary(XmlInventory) ->
+    Md5 = base64:encode(crypto:hash(md5, XmlInventory)),
```
Comment (motobob) [inline src/erlcloud_s3.erl]: is it really needed i do not see it in docs. http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTInventoryConfig.html

isn't SigV4 doing it all for you?

File: src/erlcloud_s3.erl [hunk 3]
```diff
@@ -1227,6 +1232,228 @@ set_bucket_attribute(BucketName, AttributeName, Value, Config)
...
+            {inv_key_to_name(Key), inv_encode_subtype(Value)}
+        end,
+        ListOfValues
+    );
+inv_encode_subtype({Key, Value}) ->
+    {inv_key_to_name(Key), inv_encode_subtype(Value)};
+inv_encode_subtype(Value) ->
+    [erlcloud_util:to_string(Value)].
+
+inv_key_to_name(inventory_configuration) ->  'InventoryConfiguration';
```
Comment (motobob) [inline src/erlcloud_s3.erl]: did you mean `''` or `""`? if former its unclear why we need atom2atom?

Comment (KirillDavydov) [inline src/erlcloud_s3.erl]: `' '`
atoms like `inventory_configuration` are more usable in a code while `'InventoryConfiguration'` are required by xml parser to build xml

File: src/erlcloud_s3.erl (hunk 2 above)
Comment (KirillDavydov) [inline src/erlcloud_s3.erl]: just copied that from bucket lifecycle

**Pull request merged**